pkg/cmd/cli/cmd/process.go

```diff
@@ -271,19 +275,19 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 }
 
 // injectUserVars injects user specified variables into the Template
-func injectUserVars(values []string, out io.Writer, t *templateapi.Template) {
+func injectUserVars(values []string, t *templateapi.Template) error {
 	for _, keypair := range values {
 		p := strings.SplitN(keypair, "=", 2)
 		if len(p) != 2 {
-			fmt.Fprintf(out, "invalid parameter assignment in %q: %q\n", t.Name, keypair)
-			continue
+			return fmt.Errorf("invalid parameter assignment in %q: %q\n", t.Name, keypair)
 		}
 		if v := template.GetParameterByName(t, p[0]); v != nil {
 			v.Value = p[1]
 			v.Generate = ""
 			template.AddParameter(t, *v)
 		} else {
-			fmt.Fprintf(out, "unknown parameter name %q\n", p[0])
+			return fmt.Errorf("unknown parameter name %q\n", p[0])
```

Comment:
I'm not confident we need to exit in case of providing a parameter the template is not expecting. I tend to agree with @deads2k here: not giving a value doesn't fail, invalid parameter assignment exits with a failure and giving a parameter that's not expected print to stderr but continue. @bparees how does it sound?